zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -217,25 +224,78 @@ private boolean parseSource(HttpMessage message, Source source, int depth, Strin
             resourcesfound |= processAttributeElement(message, depth, baseURL, el, "dynsrc");
         }
 
+        // Process IMPORT elements
+        elements = source.getAllElements(IMPORT_TAG);
+        for (Element el : elements) {
+            resourcesfound |=
+                    processAttributeElement(message, depth, baseURL, el, "implementation");
+        }
+
+        // Process content of container tags which hold text
+        List<String> elementsWithText =
+                Arrays.asList(
+                        HTMLElementName.P,
+                        HTMLElementName.TITLE,
+                        HTMLElementName.H1,
+                        HTMLElementName.H2,
+                        HTMLElementName.H3,
+                        HTMLElementName.H4,
+                        HTMLElementName.H5,
+                        HTMLElementName.H6,
+                        HTMLElementName.LI,
+                        HTMLElementName.BLOCKQUOTE);
+
+        for (String tag : elementsWithText) {
+            elements = source.getAllElements(tag);
+            for (Element el : elements) {
+                Source s = new Source(el.getContent());
+                if (!parseSource(message, s, depth, baseURL)) {
+                    Matcher matcher = PLAIN_COMMENTS_URL_PATTERN.matcher(s.toString());
+                    while (matcher.find()) {
+                        String foundMatch = matcher.group().trim();
+                        if (baseTagSet) {
+                            if (!baseURL.endsWith("/")) {
+                                baseURL += "/";
```

Comment:
Do not change the `baseURL` use other local variable instead, this can also be done just once before the loop.

Reply:
Done.